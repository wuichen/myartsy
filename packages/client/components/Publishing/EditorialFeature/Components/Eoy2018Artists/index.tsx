import { Box, color, Flex, media, Sans, Serif, space } from "@artsy/palette"
import { compact, find, map } from "lodash"
import React from "react"
import styled from "styled-components"

import { unica } from "Assets/Fonts"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { EditorialFeaturesProps } from "Components/Publishing/EditorialFeature/EditorialFeature"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import { ArticleCardContainer } from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCard"
import {
  ArticleCards,
  ArticlesWrapper,
} from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCards"
import {
  FullLabel,
  ImageSetContainer,
  ImageSetPreview,
  ImgContainer,
} from "Components/Publishing/Sections/ImageSetPreview"
import {
  LabelWrapper,
  SlideshowCta,
  SlideshowTitle,
} from "Components/Publishing/Sections/ImageSetPreview/ImageSetLabel"
import { StyledText } from "Components/Publishing/Sections/StyledText"
import { Text } from "Components/Publishing/Sections/Text"
import { resize } from "Utils/resizer"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export class Eoy2018Artists extends React.Component<EditorialFeaturesProps> {
  getHeaderSections = () => {
    const { sections } = this.props.article
    const headers = []

    sections.map((section, index) => {
      let headerSection
      const sectionAfter = sections[index + 1]
      const sectionAfterisImage =
        sectionAfter && sectionAfter.type === "image_collection"

      if (this.sectionIsHeader(section)) {
        headerSection = {
          index,
          section,
        }
        if (sectionAfterisImage) {
          headerSection.imageSection = sectionAfter
        }
        headers.push(headerSection)
      }
    })
    return headers
  }

  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionArtistHeader = (section, i) => {
    const headerSections = this.getHeaderSections()
    const { imageSection } = find(headerSections, ["section", section])
    const src = imageSection && imageSection.images[0].url
    const caption = imageSection && imageSection.images[0].caption

    return (
      <ArtistHeaderSection
        key={i}
        mb={40}
        flexDirection={["column", "column", "row", "row"]}
      >
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg
          src={src && resize(src, { width: 700 })}
          alignItems="flex-end"
        >
          {caption && (
            <Sans size="1" color="white" px={10} py={5}>
              <div dangerouslySetInnerHTML={{ __html: caption }} />
            </Sans>
          )}
        </ArtistHeaderImg>
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    const {
      article: { sections, layout },
      showTooltips,
    } = this.props
    const isChapterStart =
      sections[i - 1] && sections[i - 1].type === "image_collection"

    return (
      <Box maxWidth={["100%", "75%", "75%"]} ml="auto" px={[20, 0]} key={i}>
        <TextSection size="5" pb={[40, 60]} isChapterStart={isChapterStart}>
          <Text
            html={section.body}
            layout={layout}
            showTooltips={showTooltips}
          />
        </TextSection>
      </Box>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper
        key={i}
        mb={60}
        maxWidth={["100%", "100%", "100%", "75%"]}
      >
        <ImageSetPreview section={section}>
          <CaptionWrapper size={["3", "4"]}>
            <ImageSetCaption
              dangerouslySetInnerHTML={{ __html: section.images[0].caption }}
            />
          </CaptionWrapper>
        </ImageSetPreview>
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props
    const headerSections = this.getHeaderSections()
    const headerTextIndexes = map(headerSections, "index")
    const headerImages = map(headerSections, "image")

    const renderedSections = sections.map((section, i) => {
      const isHeader = headerTextIndexes.includes(i)
      const isHeaderImage = headerImages.includes(section)

      if (isHeader) {
        return this.sectionArtistHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text" && i !== 0) {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    const { article, isMobile, isTablet, isTest } = this.props
    const introText = this.sectionText(article.sections[0], 0)
    const headerImages = map(
      compact(map(this.getHeaderSections(), "imageSection")),
      "images"
    )

    return (
      <AricleWrapper>
        <Nav canFix color={color("black100")} backgroundColor="white">
          <NavBorder />
        </Nav>

        <Box px={[10, 10, 55]} maxWidth={1600} mx="auto">
          <Eoy2018ArticleHeader
            images={headerImages}
            isMobile={isMobile}
            isTablet={isTablet}
            isTest={isTest}
          />

          <ArticleContent py={40}>
            <IntroSection
              alignItems="flex-start"
              flexDirection={["column", "row"]}
              pl={[0, 20]}
            >
              <Byline article={article} />
              {introText}
            </IntroSection>
            {this.getSections()}
          </ArticleContent>
        </Box>

        {article.relatedArticles && (
          <Box px={[10, 10, 55]} mb={40} mx="auto">
            <RelatedArticleWrapper>
              <ArticleCards relatedArticles={article.relatedArticles} />
            </RelatedArticleWrapper>
          </Box>
        )}
      </AricleWrapper>
    )
  }
}

const BORDER_WIDTH = 6

const AricleWrapper = styled.div`
  ${NavContainer} {
    border-bottom: none;
  }
`

const NavBorder = styled.div`
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  position: absolute;
  top: 100%;
  left: 55px;
  right: 55px;
  max-width: calc(1600px - 110px);
  margin: 0 auto;

  ${media.md`
    left: 10px;
    right: 10px;
  `};
`

const ArticleContent = styled(Box)`
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
    line-height: 1.3em;
    margin: 0;
    font-weight: inherit;

    ${media.sm`
      ${unica("s25")};
      line-height: 1.3em;
    `};
  }
`

export const ArtistHeaderSection = styled(Flex)`
  height: 60vh;
  min-height: 450px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left-width: 0;

  ${media.md`
    height: fit-content;
  `};
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  overflow: hidden;
  min-height: min-content;

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    min-height: fit-content;
    padding: ${space(2)}px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    margin: 0;
    font-weight: inherit;

    ${media.md`
      width: 60%;
      float: right;
      height: 100%;
      border-bottom: none;
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};

    ${media.xs`
      ${unica("s45")};
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
      border-left: none;
      width: 100%;
      height: 50%;
      float: none;
    `};
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    min-height: fit-content;
    display: inline-flex;
    padding: ${space(2)}px;
    margin: 0;
    font-weight: inherit;

    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }

    ${media.md`
      width: 40%;
      float: left;
      &:last-child {
        border-left: none;
        border-top: ${BORDER_WIDTH}px solid ${color("purple100")};
      }
    `};

    ${media.xs`
      ${unica("s19")};
      width: 50%;
      float: none;
      &:last-child {
        border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
        border-top: none;
      }
    `};
  }

  ${media.md`
    height: 65vh;
    border-right: none;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ArtistHeaderImg = styled(Flex)<{ src?: string }>`
  flex: 1;
  background: ${color("purple100")};

  p {
    text-shadow: 0 0 5px black;
    opacity: 0.6;
  }

  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};

  ${media.md`
    min-height: 70vw;
  `};
`

const IntroSection = styled(Flex)`
  ${BylineContainer} {
    flex-direction: column;
    align-items: flex-start;
    ${media.md`
      padding: 0 20px 20px;
    `};
  }
`

const TextSection = styled(Serif)<{ isChapterStart?: boolean }>`
  ${StyledText} {
    blockquote {
      padding: 0;
    }

    p,
    .paragraph {
      font-size: 24px;
      text-indent: 2em;
      padding: 0;

      ${props =>
        props.isChapterStart &&
        `
        &:first-child {
          text-indent: 0;
        }
    `};
    }
  }
`

const CaptionWrapper = styled(Sans)`
  flex: 1;

  ${media.sm`
    min-width: 50%;
  `};
`

const ImageSetCaption = styled.div`
  height: 100%;
  padding: ${space(2)}px;
  background: white;
  color: black;
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${media.sm`
    border-bottom: none;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ImageSetWrapper = styled(Box)`
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;

  ${ImageSetContainer} {
    display: flex;
    flex-direction: row-reverse;

    ${media.sm`
      flex-direction: column-reverse;
    `};
  }

  ${FullLabel} {
    position: relative;
    display: flex;
    flex-direction: column-reverse;
    flex: 1;
    left: auto;
    bottom: auto;
    max-width: inherit;
    border-radius: 0;
    box-shadow: none;

    &:hover {
      background-color: ${color("purple100")};
    }
    ${media.sm`
      flex-direction: row-reverse;

      ${LabelWrapper} {
        max-width: 50%;
      }
    `};
  }

  ${SlideshowTitle} {
    display: none;
  }

  ${SlideshowCta} {
    flex-direction: column;

    > div {
      padding: 0;
    }
  }

  ${ImgContainer} {
    flex: 3;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

    img {
      object-fit: cover;
      object-position: center;
      height: 100%;
    }

    ${media.sm`
      border-right: none;
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};
  }
`

const RelatedArticleWrapper = styled.div`
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${ArticlesWrapper} {
    margin-bottom: 0;
    margin-top: 0;
  }

  ${ArticleCardContainer} {
    border: none;
  }
`
