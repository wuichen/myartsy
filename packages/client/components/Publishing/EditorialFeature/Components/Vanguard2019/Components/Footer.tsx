import { Box, Flex, Sans, Separator, Serif } from "@artsy/palette"
import React from "react"

export const VanguardFooter = () => {
  return (
    <Box background="white" pb={80}>
      <Box
        maxWidth={["100vw", 1000]}
        px={["10vw", "10vw", "10vw", "10vw", 4]}
        mx="auto"
      >
        <Separator mb={4} />
        <Flex flexDirection={["column", "row"]}>
          <Box pr={[0, 2]} width={["100%", "50%"]} pb={4}>
            <Sans size="3" weight="medium">
              Corrections
            </Sans>
            <Serif size="3" element="p">
              {corrections}
            </Serif>
          </Box>
          <Box pl={[0, 2]} width={["100%", "50%"]} pb={4}>
            <Sans size="3" weight="medium">
              Clarifications
            </Sans>
            <Serif size="3" element="p">
              {clarifications}
            </Serif>
          </Box>
        </Flex>
        <Separator mb={2} />
        <Serif size="3">
          Back to <a href="/articles">Artsy Editorial</a>
        </Serif>
      </Box>
    </Box>
  )
}

const corrections = `A previous version of this article referred to Aike gallery as
Aike-Dellarco; McArthur Binion’s show at the Mississippi Museum of
Art, originally scheduled for this fall, is not opening until 2020;
and Melike Kara is Kurdish-German not Turkish-German. The text has
been updated to reflect these changes.`

const clarifications = `While only Beyoncé was named as the buyer of a Derek Fordjour work at
Frieze New York, Jay-Z was actually the buyer; in order to fully
recognize the breadth of galleries that represent the featured
artists, additional gallery names have been added to the text for
Beatriz González, Vivian Suter, and Diane Simpson; while the text on
Henry Taylor previously referred to Eva Presenhuber as his dealer, it
has been updated to reflect that Blum & Poe has been primarily
representing the artist for the past decade; and mention of
SculptureCenter, organizers of Tishan Hsu’s 2020 show at the Hammer,
has been added to clarify their involvement with the exhibition.`
