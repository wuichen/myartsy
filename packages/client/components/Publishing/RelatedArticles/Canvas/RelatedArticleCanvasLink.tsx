import { Serif } from "@artsy/palette"
import { track } from "Artsy/Analytics"
import * as Schema from "Artsy/Analytics/Schema"
import { RelatedArticleCanvasData } from "Components/Publishing/Typings"
import React from "react"
import styled from "styled-components"
import { crop } from "../../../../Utils/resizer"
import { pMedia } from "../../../Helpers"
import { Byline } from "../../Byline/Byline"
import { getEditorialHref } from "../../Constants"

interface RelatedArticleCanvasLinkProps
  extends React.HTMLProps<HTMLDivElement> {
  article: RelatedArticleCanvasData
}

@track()
export class RelatedArticleCanvasLink extends React.Component<
  RelatedArticleCanvasLinkProps
> {
  @track<RelatedArticleCanvasLinkProps>(props => ({
    action_type: Schema.ActionType.Click,
    destination_path: getEditorialHref(
      props.article.layout,
      props.article.slug
    ),
    type: Schema.Type.Thumbnail,
  }))
  onClick() {
    // noop
  }

  render() {
    const { article } = this.props
    const href = getEditorialHref(article.layout, article.slug)
    const imageSrc = crop(article.thumbnail_image, { width: 510, height: 340 })
    const bylineArticle = { ...article, id: article.slug }

    return (
      <ArticleFigure href={href} onClick={this.onClick.bind(this)}>
        <ImageTitle>
          <BlockImage src={imageSrc} alt={article.thumbnail_title} />
          <Serif size="3t">{article.thumbnail_title}</Serif>
        </ImageTitle>

        <Byline article={bylineArticle} layout="condensed" />
      </ArticleFigure>
    )
  }
}

const ImageTitle = styled.div`
  display: flex;
  flex-direction: column;
  color: black;
  text-decoration: none;
`

const ArticleFigure = styled.a`
  display: flex;
  flex-direction: column;
  width: 278px;
  text-decoration: none;
  color: black;
  ${pMedia.sm`width: 225px;`};
`

const BlockImage = styled.img`
  display: block;
  width: 278px;
  height: 185px;
  margin-bottom: 10px;
  object-fit: cover;
  ${pMedia.sm`
    width: 225px;
    height: 150px;
  `};
`
