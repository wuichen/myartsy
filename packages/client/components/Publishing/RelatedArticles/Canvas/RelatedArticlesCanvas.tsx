import { Flex, Sans } from "@artsy/palette"
import { track } from "Artsy/Analytics"
import * as Schema from "Artsy/Analytics/Schema"
import { RelatedArticleCanvasData } from "Components/Publishing/Typings"
import { map, once } from "lodash"
import React from "react"
import Waypoint from "react-waypoint"
import styled from "styled-components"
import { pMedia } from "../../../Helpers"
import { RelatedArticleCanvasLink } from "./RelatedArticleCanvasLink"

export interface RelatedArticlesCanvasProps
  extends React.HTMLProps<HTMLDivElement> {
  vertical?: {
    name: string
    id?: string
  }
  articles: RelatedArticleCanvasData[]
  isMobile?: boolean
}

interface ScrollingContainerProps {
  isMobile?: boolean
}

@track({
  // TODO: re-evalutate double naming of context/schema
  context_module: Schema.ContextModule.FurtherReading,
  subject: Schema.Subject.FurtherReading,
})
export class RelatedArticlesCanvas extends React.Component<
  RelatedArticlesCanvasProps
> {
  @track(() => ({ action_type: Schema.ActionType.Impression }))
  trackRelatedImpression() {
    // noop
  }

  render() {
    const { articles, isMobile, vertical } = this.props

    return (
      <Flex flexDirection="column" maxWidth={1250} mx="auto" mt={4} mb={6}>
        {getTitle(vertical)}
        <Waypoint onEnter={once(this.trackRelatedImpression.bind(this))} />
        <ArticlesWrapper isMobile={isMobile}>
          {map(articles, (article, i) => {
            return (
              <RelatedArticleCanvasLink
                article={article}
                key={`related-article-figure-${i}`}
              />
            )
          })}
        </ArticlesWrapper>
      </Flex>
    )
  }
}

const getTitle = vertical => {
  if (vertical) {
    return (
      <Title size={["6", "8"]}>
        Further reading in <VerticalSpan>{vertical.name}</VerticalSpan>
      </Title>
    )
  } else {
    return <Title size={["6", "8"]}>More from Artsy Editorial</Title>
  }
}

const Title = styled(Sans)`
  margin-bottom: 20px;
  ${pMedia.xl`
    margin: 0 20px 30px 40px;
  `};
  ${pMedia.sm`
    margin-left: 20px;
  `};
`
const VerticalSpan = styled.span`
  ${pMedia.sm`
    display: block;
  `};
`
const ArticlesWrapper = styled.div.attrs<ScrollingContainerProps>({})`
  display: flex;
  justify-content: space-between;
  overflow-x: scroll;

  ::-webkit-scrollbar {
    display: none;
  }

  a {
    margin-right: 30px;

    &:last-child {
      margin-right: 0;
    }
    ${pMedia.xl`
      margin: 0 10px;
      &:first-child {
        margin-left: 40px;
      }
      &:last-child {
        border-right: 20px solid white;
      }
    `};
    ${pMedia.sm`
      &:first-child {
        margin-left: 20px;
      }
    `};
  }
  ${props => props.isMobile && "-webkit-overflow-scrolling: touch;"};
`
