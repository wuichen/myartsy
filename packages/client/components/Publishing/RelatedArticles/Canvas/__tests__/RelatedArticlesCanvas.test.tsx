import { mockTracking } from "Artsy/Analytics"
import { RelatedCanvas } from "Components/Publishing/Fixtures/Components"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import Waypoint from "react-waypoint"
import { RelatedArticleCanvasLink } from "../RelatedArticleCanvasLink"
import { RelatedArticlesCanvas } from "../RelatedArticlesCanvas"

jest.unmock("react-tracking")

describe("RelatedArticlesCanvas", () => {
  const getWrapper = props => {
    return mount(<RelatedArticlesCanvas {...props} />)
  }

  let testProps
  beforeEach(() => {
    testProps = { articles: RelatedCanvas, vertical: { name: "Art Market" } }
  })

  it("renders the related articles canvas", () => {
    const related = renderer
      .create(<RelatedArticlesCanvas {...testProps} />)
      .toJSON()
    expect(related).toMatchSnapshot()
  })

  it("renders the vertical name if there is one", () => {
    const component = getWrapper(testProps)
    expect(component.text()).toMatch("Further reading in Art Market")
  })

  it("renders a default message if there is no vertical", () => {
    delete testProps.vertical
    const component = getWrapper(testProps)
    expect(component.html()).toMatch("More from Artsy Editorial")
  })

  it("renders article links", () => {
    const component = getWrapper(testProps)
    expect(component.find(RelatedArticleCanvasLink)).toHaveLength(4)
  })

  it("Calls a tracking impression", () => {
    const { Component, dispatch } = mockTracking(RelatedArticlesCanvas)
    const component = mount(<Component articles={RelatedCanvas} />)
    component
      .find(Waypoint)
      .getElement()
      .props.onEnter()

    expect(dispatch).toBeCalledWith({
      action_type: "Impression",
      context_module: "Further reading",
      subject: "Further reading",
    })
  })

  it("Tracks link clicks", () => {
    const { Component, dispatch } = mockTracking(RelatedArticlesCanvas)
    const component = mount(<Component articles={RelatedCanvas} />)
    component
      .find(RelatedArticleCanvasLink)
      .at(0)
      .simulate("click")

    expect(dispatch).toBeCalledWith({
      action_type: "Click",
      context_module: "Further reading",
      subject: "Further reading",
      destination_path:
        "/article/artsy-editorial-15-top-art-schools-united-states",
      type: "thumbnail",
    })
  })
})
