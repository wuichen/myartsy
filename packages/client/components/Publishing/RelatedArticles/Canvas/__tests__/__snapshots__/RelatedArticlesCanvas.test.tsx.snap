// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedArticlesCanvas renders the related articles canvas 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 1250px;
  margin-bottom: 128px;
  margin-top: 32px;
  margin-left: auto;
  margin-right: auto;
}

.c2 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 22px;
  line-height: 30px;
}

.c8 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 16px;
  line-height: 20px;
}

.c10 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
}

.c11 {
  margin: 10px 20px 0 0;
}

.c11::before {
  content: "";
  display: inline-block;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 0 5px 1px 0;
  background-color: black;
}

.c12 {
  margin: 5px 20px 0 0;
  white-space: nowrap;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  color: black;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 278px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

.c7 {
  display: block;
  width: 278px;
  height: 185px;
  margin-bottom: 10px;
  object-fit: cover;
}

.c1 {
  margin-bottom: 20px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  overflow-x: scroll;
}

.c4::-webkit-scrollbar {
  display: none;
}

.c4 a {
  margin-right: 30px;
}

.c4 a:last-child {
  margin-right: 0;
}

@media screen and (min-width:40em) {
  .c2 {
    font-size: 28px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    line-height: 36px;
  }
}

@media (max-width:720px) {
  .c5 {
    width: 225px;
  }
}

@media (max-width:720px) {
  .c7 {
    width: 225px;
    height: 150px;
  }
}

@media (max-width:1280px) {
  .c1 {
    margin: 0 20px 30px 40px;
  }
}

@media (max-width:720px) {
  .c1 {
    margin-left: 20px;
  }
}

@media (max-width:720px) {
  .c3 {
    display: block;
  }
}

@media (max-width:1280px) {
  .c4 a {
    margin: 0 10px;
  }

  .c4 a:first-child {
    margin-left: 40px;
  }

  .c4 a:last-child {
    border-right: 20px solid white;
  }
}

@media (max-width:720px) {
  .c4 a:first-child {
    margin-left: 20px;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
    fontSize={
      Array [
        "22px",
        "28px",
      ]
    }
  >
    Further reading in 
    <span
      className="c3"
    >
      Art Market
    </span>
  </div>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
  <div
    className="c4"
  >
    <a
      className="c5"
      href="/article/artsy-editorial-15-top-art-schools-united-states"
      onClick={[Function]}
    >
      <div
        className="c6"
      >
        <img
          alt="The 15 Top Art Schools in the United States"
          className="c7"
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F4Tq-iYkN8dOpshFoKRXyYw%252Fcustom-Custom_Size___PoetterHall_Exterior%2Bcopy.jpg&width=510&height=340&quality=95"
        />
        <div
          className="c8"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize="16px"
        >
          The 15 Top Art Schools in the United States
        </div>
      </div>
      <div
        className="Byline c9"
        color="black"
      >
        <div
          className="c10"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="12px"
        >
          <div
            className="c11"
            color="black"
          >
            Anna Louis-Sussman and Kana Abe
          </div>
        </div>
        <div
          className="c12"
        >
          <div
            className="c10"
            fontFamily={
              Object {
                "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                "fontWeight": 500,
              }
            }
            fontSize="12px"
          >
            May 19, 2017
          </div>
        </div>
      </div>
    </a>
    <a
      className="c5"
      href="/article/artsy-editorial-four-years-walter-de-marias-death-final-work-complete"
      onClick={[Function]}
    >
      <div
        className="c6"
      >
        <img
          alt="Four Years after Walter De Maria’s Death, His Final Work Is Complete"
          className="c7"
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F6IqxBTQCkExip2auQ7ZWCA%252FDEMAR-2011.0006-B.jpg&width=510&height=340&quality=95"
        />
        <div
          className="c8"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize="16px"
        >
          Four Years after Walter De Maria’s Death, His Final Work Is Complete
        </div>
      </div>
      <div
        className="Byline c9"
        color="black"
      >
        <div
          className="c10"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="12px"
        >
          <div
            className="c11"
            color="black"
          >
            Halley Johnson
          </div>
        </div>
        <div
          className="c12"
        >
          <div
            className="c10"
            fontFamily={
              Object {
                "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                "fontWeight": 500,
              }
            }
            fontSize="12px"
          >
            May 19, 2017
          </div>
        </div>
      </div>
    </a>
    <a
      className="c5"
      href="/series/artsy-editorial-french-art-history-in-a-nutshell"
      onClick={[Function]}
    >
      <div
        className="c6"
      >
        <img
          alt="French Art History in a Nutshell"
          className="c7"
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FlEcCm2XbfZ7bPAVgLlM21w%252Flarger-21.jpg&width=510&height=340&quality=95"
        />
        <div
          className="c8"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize="16px"
        >
          French Art History in a Nutshell
        </div>
      </div>
      <div
        className="Byline c9"
        color="black"
      >
        <div
          className="c10"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="12px"
        >
          <div
            className="c11"
            color="black"
          >
            Casey Lesser
          </div>
        </div>
        <div
          className="c12"
        >
          <div
            className="c10"
            fontFamily={
              Object {
                "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                "fontWeight": 500,
              }
            }
            fontSize="12px"
          >
            May 19, 2017
          </div>
        </div>
      </div>
    </a>
    <a
      className="c5"
      href="/news/artsy-editorial-miami-artists-museums-brace-hurricane-irma"
      onClick={[Function]}
    >
      <div
        className="c6"
      >
        <img
          alt="Miami Artists and Museums Brace for Hurricane Irma"
          className="c7"
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FjAu4NaKnr_m53OnnMaDe_w%252Fmag.jpg&width=510&height=340&quality=95"
        />
        <div
          className="c8"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize="16px"
        >
          Miami Artists and Museums Brace for Hurricane Irma
        </div>
      </div>
      <div
        className="Byline c9"
        color="black"
      >
        <div
          className="c10"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="12px"
        >
          <div
            className="c11"
            color="black"
          >
            Owen Dodd
          </div>
        </div>
        <div
          className="c12"
        >
          <div
            className="c10"
            fontFamily={
              Object {
                "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                "fontWeight": 500,
              }
            }
            fontSize="12px"
          >
            May 19, 2017
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
